server/text_generation_server/utils/tokens.py

```diff
@@ -78,13 +98,20 @@ def __call__(self, input_ids, scores):
 
         next_id = self.choice(scores[-1]).view(1, 1)
 
+        if self.grammar_processor is not None:
+            next_state = self.grammar_processor.advance(
+                next_id.item(), self.fsm_grammar_state, self.grammar
```

Comment:
Same remark here for the `.item()` although here it's less critical since it's already calling all the processors serially which is quite slow already.